src/objectExplorer/objectExplorerService.ts

```diff
@@ -133,10 +133,9 @@ export class ObjectExplorerService {
         const handler = (result: ExpandResponse) => {
             if (result && result.nodes) {
                 const credentials = self._sessionIdToConnectionCredentialsMap.get(result.sessionId);
-                let parentNode = self._rootTreeNodeArray.find(n => n.nodePath === result.nodePath);
                 const children = result.nodes.map(node => TreeNodeInfo.fromNodeInfo(node, result.sessionId,
-                    parentNode, credentials));
-                self._treeNodeToChildrenMap.set(parentNode, children);
+                    self._currentNode, credentials));
```

Comment:
Need to set `self._currentNode` because that's the selected node when it's expanded. Looking into just rootNodeArray would fail as we keep expanding and keep going deeper.

Reply:
I still really don't like storing _currentNode like this. What exactly is preventing a race condition with the _currentNode where simultaneous actions update the _currentNode before something like this has finished executing?

Reply:
because it's set right when a node is clicked on and hence the `await` will wait for the notification handler to finish doing it work before setting the next `_currentNode`